Controller/Adminhtml/Importcategory/Save.php

```diff
@@ -80,138 +80,165 @@ public function execute()
-                        if ($parent_cat && $parentid) {
-                            $exist_categories_pathname[$parentid] = $parent_cat;
+                        
+                        $parent_cat = implode('/', $parentCatNames);
+                        
+                        if ($parent_cat && $parentId) {
+                            $existCategoriesPathName[$parentId] = $parent_cat;
                         }
                     }
-                    foreach ($data as $key => $categoryitem) {
+
+                    foreach ($data as $key => $categoryItem) {
+
                         if ($key!=0) {
-                            $cat_data = $this->_getKeyValue($categoryitem, $header);
+                            $cat_data = $this->_getKeyValue($categoryItem, $header);
+
                             if (isset($cat_data['category_id'])) {
                                 unset($cat_data['category_id']);
                             }
-                            if (isset($categorieskey) && ($categorieskey!='' || $categorieskey===0)) {
-                                $array_key = array_search($categoryitem[$categorieskey], $exist_categories_pathname);
+
+                            if (isset($categoriesKey) && ($categoriesKey!='' || $categoriesKey===0)) {
+                                $array_key = array_search($categoryItem[$categoriesKey], $existCategoriesPathName);
+
                                 if ($array_key) {
-                                    $alreadyexist[] = $categoryitem[$categorieskey];
-                                } else {
-                                    $strmark = strrpos($categoryitem[$categorieskey], '/');
-                                    $_parentid = '';
-                                    $newcategory = '';
+                                    $alreadyExist[] = $categoryItem[$categoriesKey];
+                                }
+
+                                else {
+                                    $strmark = strrpos($categoryItem[$categoriesKey], '/');
+                                    $_parentId = '';
+                                    $newCategory = '';
+
                                     if ($strmark!=false) {
-                                        $parentpath = substr($categoryitem[$categorieskey], 0, ($strmark));
-                                        $newcategory = substr($categoryitem[$categorieskey], ($strmark)+1);
-                                        $_parentid = array_search($parentpath, $exist_categories_pathname);
-                                    } else {
-                                        $newcategory = $categoryitem[$categorieskey];
-                                        $_parentid = $cat_info->getId();
+                                        $parentpath = substr($categoryItem[$categoriesKey], 0, ($strmark));
+                                        $newCategory = substr($categoryItem[$categoriesKey], ($strmark)+1);
```

Comment:
please add space before `+` and are the brackets around `$strmark` necessary 

should `$strmark` be `$strMark`?